src/map/script.c

```diff
@@ -7437,52 +7437,136 @@ BUILDIN(grouprandomitem) {
 }
 
 /*==========================================
- *
- *------------------------------------------*/
-BUILDIN(makeitem)
-{
-	int nameid,amount;
-	int x,y,m;
+* makeitem <item id or name>,<amount>,"<map name>",<X>,<Y>;
+*------------------------------------------*/
+BUILDIN(makeitem) {
+	int nameid, amount;
+	int x, y, m;
 	const char *mapname;
 	struct item item_tmp;
 
-	if( script_isstringtype(st, 2) ) {
+	if (script_isstringtype(st, 2)) {
 		const char *name = script_getstr(st, 2);
 		struct item_data *item_data = itemdb->search_name(name);
 		if (item_data)
 			nameid = item_data->nameid;
 		else
 			nameid = UNKNOWN_ITEM_ID;
-	} else {
+	}
+	else {
 		nameid = script_getnum(st, 2);
-		if( nameid <= 0 || !itemdb->exists(nameid)) {
+		if (nameid <= 0 || !itemdb->exists(nameid)) {
 			ShowError("makeitem: Nonexistant item %d requested.\n", nameid);
 			return false; //No item created.
 		}
 	}
-	amount  = script_getnum(st,3);
-	mapname = script_getstr(st,4);
-	x       = script_getnum(st,5);
-	y       = script_getnum(st,6);
+	amount = script_getnum(st, 3);
+	mapname = script_getstr(st, 4);
+	x = script_getnum(st, 5);
+	y = script_getnum(st, 6);
 
-	if(strcmp(mapname,"this")==0) {
+	if (strcmp(mapname, "this") == 0) {
 		TBL_PC *sd;
 		sd = script->rid2sd(st);
 		if (!sd) return true; //Failed...
-		m=sd->bl.m;
-	} else
-		m=map->mapname2mapid(mapname);
+		m = sd->bl.m;
+	}
+	else
+		m = map->mapname2mapid(mapname);
 
-	if( m == -1 ) {
+	if (m == -1) {
 		ShowError("makeitem: creating map on unexistent map '%s'!\n", mapname);
 		return false;
 	}
 
-	memset(&item_tmp,0,sizeof(item_tmp));
+	memset(&item_tmp, 0, sizeof(item_tmp));
 	item_tmp.nameid = nameid;
-	item_tmp.identify=1;
+	item_tmp.identify = 1;
+
+	if (!itemdb->isstackable(nameid))
+		amount = 1;
+
+	script_pushint(st, map->addflooritem(NULL, &item_tmp, amount, m, x, y, 0, 0, 0, 0) > 0);
```

Comment:
value of true/false returned by this check should be 1/0 however still it's not safe value please change this line to something like (map->addflooritem(NULL, &item_tmp, amount, m, x, y, 0, 0, 0, 0) == 1)? 1 : 0